FirmataMarshaller.cpp

```diff
@@ -168,6 +168,8 @@ const
   if ( (Stream *)NULL == FirmataStream ) { return; }
   FirmataStream->write(START_SYSEX);
   FirmataStream->write(REPORT_FIRMWARE);
+  FirmataStream->write(firmata::FIRMWARE_MAJOR_VERSION);
```

Comment:
Maybe what is needed here instead is two new Firmata commands `QUERY_FIRMWARE_VERSION` and `QUERY_PROTOCOL_VERSION` and those have no payload. Only the Reply has data.